File: scripts/algorithms/M/Maximum Number of Consecutive Values You Can Make/Maximum Number of Consecutive Values You Can Make.py
// Runtime: 576 ms (Top 97.83%) | Memory: 23.00 MB (Top 5.22%)

class Solution:
    def getMaximumConsecutive(self, coins: List[int]) -> int:
        cur_max = 0
        coins.sort()
      
        for coin in coins:
            if coin == 1:
                cur_max += 1
            elif coin <= cur_max+1:
                cur_max += coin
            else: #coin > cur_max + 1
                break
        
        return cur_max+1

File: scripts/algorithms/M/Maximum Number of Consecutive Values You Can Make/Maximum Number of Consecutive Values You Can Make.cpp
class Solution {
public:
    int getMaximumConsecutive(vector<int>& coins) {
        sort(coins.begin(), coins.end());
        int n = coins.size();
        int ans = 1;
        for(int i = 0; i < n; i++) {
            if(coins[i] > ans) return ans;
            ans += coins[i]; 
        }
        return ans;
    }
};